### PR TITLE
Google maps listing markers

### PR DESCRIPTION
**Pull request by Pochetes**
## Maps Listing Markers
- made custom map markers that display the listing's price upfront
- able to click on the map marker to see a preview of the listing, can click on the listing's image to see further details about the listing
- fixed bug where initialization of Google Maps API had unnecessary renders. Moved the loading script to root of project where both the Autocomplete widget and the map loaded once.

### Unrelated
- created a wrapper component that redirects non-authenticated users to the `/login` route and to `/menu` if they are
- fixed `Unauthorized` errors when first loading website data

## Demo
the user is able to see a map marker of all the listings around the area, displaying the price upfront, like so

Screen Shot 2022-08-07 at 5 16 40 PM

when they click on one of the map markers, they can see a preview of the listing and can also click on it to straight to the listing

Screen Shot 2022-08-07 at 5 16 55 PM

`NOTE` the commit "added basic search implementation" is a duplicate commit of the previous diff. All I added was a comment as suggested.

### CODE REVIEW

File: swimster-ui/src/hooks/usePoolMap.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    const [selected, setSelected] = useState({})
+    const [userLocation, setUserLocation] = useState({})
+    const [error, setError] = useState({})
+
+    const mapContainerStyle = {
+        width: '100%',
+        height: '100%',
+    }
+
+    const defaultCenter = {
```
Comment (Pochetes) [inline swimster-ui/src/hooks/usePoolMap.js:15]: need to the add the comment here

**Review by roeshanigos — APPROVED** [review]
pretty technically sound

just some design considerations: 
- would be nice for the user to carousel on the hover cards on the map (on top of the listings page!) 
- maybe we should toggle between map vie and list view so its not all in the same landing page view

File: swimster-ui/src/components/PoolMap/PoolMap.jsx [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
+                })}
+                {selected.title && (
+                    <InfoWindow
+                    position={{ lat: parseFloat(selected.lat), lng: parseFloat(selected.lng)}}
+                    clickable={true}
+                    onCloseClick={() => setSelected({})}
+                    >
+                        <div style={infoWindowContainer}>
+                            <Link to={`/listings/${selected.id}`}>
+                            <img style={poolOptions.listingImageOnMap} src={selected?.image_url} alt="pool listing pic on map" />
```
[the remark this answers is not in the document]
Comment (Pochetes) [reply swimster-ui/src/components/PoolMap/PoolMap.jsx:45]: done!

File: swimster-ui/src/hooks/usePoolMap.js (hunk 1 above)
Comment (Pochetes) [reply swimster-ui/src/hooks/usePoolMap.js:15]: done

**Pull request merged**